Packs/AzureKeyVault/Integrations/AzureKeyVault/AzureKeyVault.py

```diff
@@ -1107,35 +1204,94 @@ def get_certificate_policy_command(client: KeyVaultClient, args: Dict[str, Any])
+                                      ['subscriptionId', 'tenantId',
+                                       'state', 'displayName'
+                                       ],
+                                      removeNull=True, headerTransform=string_to_table_header)
+    return CommandResults(
+        outputs_prefix='AzureKeyVault.Subscription',
+        outputs_key_field='id',
+        outputs=outputs,
+        raw_response=response,
+        readable_output=readable_output,
+    )
+
+
+def list_resource_groups_command(client: KeyVaultClient, args: Dict[str, Any], params: Dict[str, Any]) -> CommandResults:
+    """
+    List all resource groups in the subscription.
+
+    Args:
+        client (KeyVaultClient):  Azure Key Vault API client.
+        args (Dict[str, Any]): command arguments.
+        params (Dict[str, Any]): configuration parameters.
+
+    Returns:
+        CommandResults: Command results with raw response, outputs and readable outputs.
+
+    """
+    tag = args.get('tag')
+    limit = arg_to_number(args.get('limit', DEFAULT_LIMIT))
+    subscription_id_list = argToList(args.get('subscription_id', params.get('subscription_id')))
+
+    all_responses = []
+    for subscription_id in subscription_id_list:
+        response = client.list_resource_groups_request(subscription_id=subscription_id, tag=tag, limit=limit)
+        if "error" in response:
+            raise Exception(response)
+        all_responses.extend(response)
+
+    readable_output = tableToMarkdown('Resource Groups List',
+                                      all_responses,
+                                      ['name', 'location', 'tags',
+                                       'properties.provisioningState'
+                                       ],
+                                      removeNull=True, headerTransform=string_to_table_header)
+    return CommandResults(
+        outputs_prefix='AzureKeyVault.ResourceGroup',
+        outputs_key_field='id',
+        outputs=all_responses,
+        raw_response=all_responses,
+        readable_output=readable_output,
+    )
+
+
+def test_module(client: KeyVaultClient, params: Dict[str, any]) -> None:
     """
      Test instance parameters validity.
      Displays an appropriate message in case of invalid parameters.
 
      Args:
          client (KeyVaultClient):  Azure Key Vault API client.
+         params (Dict[str, Any]): configuration parameters.
 
      Returns:
          None
      """
     try:
+        subscription_id = params.get('subscription_id')
         client.ms_client.get_access_token(resource=MANAGEMENT_RESOURCE)
         client.ms_client.get_access_token(resource=VAULT_RESOURCE)
-        client.list_key_vaults_request(1, 0)
-        # fetch_credentials(client,[''],[''],'xsoar-test-vault/test-sec-1')
+        client.list_key_vaults_request(subscription_id=subscription_id, limit=1, offset=0)
         return_results('ok')
 
-    except Exception as error:
```

Comment:
Did we move it to the main func cause now the subscription-related errors might be raised in many commands (due to the new subscription arg)?

Reply:
Yes